api/equality/equality.go

```diff
@@ -27,3 +30,50 @@ func TaskStatusesEqualStable(a, b *api.TaskStatus) bool {
 	copyA.Timestamp, copyB.Timestamp = nil, nil
 	return reflect.DeepEqual(&copyA, &copyB)
 }
+
+// RootCAEqualStable compares RootCAs, excluding join tokens, which are randomly generated
+func RootCAEqualStable(a, b *api.RootCA) bool {
+	switch {
+	case a == nil && b == nil:
+		return true
+	case a != nil && b != nil:
+	default:
+		return false
+	}
+	if !(bytes.Equal(a.CACert, b.CACert) &&
+		subtle.ConstantTimeCompare(a.CAKey, b.CAKey) == 1 &&
+		a.CACertHash == b.CACertHash) {
+		return false
+	}
+	switch {
+	case a.RootRotation == nil && b.RootRotation == nil:
+		return true
+	case a.RootRotation != nil && b.RootRotation != nil:
+	default:
+		return false
+	}
+	return (bytes.Equal(a.RootRotation.CACert, b.RootRotation.CACert) &&
+		subtle.ConstantTimeCompare(a.RootRotation.CAKey, b.RootRotation.CAKey) == 1 &&
+		bytes.Equal(a.RootRotation.CrossSignedCACert, b.RootRotation.CrossSignedCACert))
+}
+
+// ExternalCAsEqualStable compares lists of external CAs and determines whether they are equal
+func ExternalCAsEqualStable(a, b []*api.ExternalCA) bool {
```

Comment:
We don't actually necessarily want a `reflect.DeepEqual` of the whole struct - `reflect.DeepEqual` doesn't treat nil and empty values the same:  https://play.golang.org/p/VGgZq6MVik

I figure an empty list of external CA should be the same, for our purposes, as a nil list.  It also won't treat an empty map the same as a nil map (and the `api.ExternalCA` object has a map of options).